File: кт5/test1.py
import pytest
import requests

BASE_URL = "https://petstore.swagger.io/v2"
USER_URL = f"{BASE_URL}/user"
STORE_URL = f"{BASE_URL}/store/order"


@pytest.fixture
def create_user():
    user_data = {
        "id": 123,
        "username": "testuser",
        "firstName": "Test",
        "lastName": "User",
        "email": "testuser@example.com",
        "password": "password123",
        "phone": "1234567890",
        "userStatus": 1
    }
    response = requests.post(USER_URL, json=user_data)
    assert response.status_code == 200, "User creation failed"
    return user_data


def test_create_user(create_user):
    response = requests.get(f"{USER_URL}/{create_user['username']}")
    assert response.status_code == 200, "User not found after creation"
    data = response.json()
    assert data['username'] == create_user['username'], "User data mismatch"




@pytest.fixture
def create_order():
    order_data = {
        "id": 1,
        "petId": 100,
        "quantity": 2,
        "shipDate": "2023-10-31T00:00:00.000Z",
        "status": "placed",
        "complete": True
    }
    response = requests.post(STORE_URL, json=order_data)
    assert response.status_code == 200, "Order creation failed"
    return order_data


def test_get_order(create_order):
    response = requests.get(f"{STORE_URL}/{create_order['id']}")
    assert response.status_code == 200, "Order not found"
    data = response.json()
    assert data['status'] == create_order['status'], "Order status mismatch"


def test_delete_order(create_order):
    response = requests.delete(f"{STORE_URL}/{create_order['id']}")
    assert response.status_code == 200, "Failed to delete order"
    
    response = requests.get(f"{STORE_URL}/{create_order['id']}")
    assert response.status_code == 404, "Order still exists after deletion"

File: кт6/testsapi/Tests/Test1.cs
using System.Net.Http;
using System.Threading.Tasks;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using Newtonsoft.Json;
using System.Threading;

[TestClass]
public class PetStoreTests
{
    private PetStoreClient _client;

    [TestInitialize]
    public void Setup()
    {
        _client = new PetStoreClient();
    }

    [TestMethod]
    [TestCategory("User Management")]
    [Description("Create and retrieve a user")]
    public async Task TestCreateUser()
    {
        var user = new UserModel
        {
            Id = 0,
            Username = "string",
            FirstName = "string",
            LastName = "string",
            Email = "string",
            Password = "string",
            Phone = "string",
            UserStatus = 0
        };

        var createResponse = await _client.CreateUserAsync(user);
        Assert.AreEqual(200, (int)createResponse.StatusCode, "User creation failed");

        Thread.Sleep(1000); 

        var getResponse = await _client.GetUserAsync(user.Username);
        Assert.AreEqual(200, (int)getResponse.StatusCode, "User not found after creation");

        var getUser = JsonConvert.DeserializeObject<UserModel>(await getResponse.Content.ReadAsStringAsync());
        Assert.AreEqual(user.Username, getUser.Username, "User data mismatch");
    }

    [TestMethod]
    [TestCategory("Order Management")]
    [Description("Retrieve an order")]
    public async Task TestGetOrder()
    {
        var order = new OrderModel
        {
            Id = 12,
            PetId = 0,
            Quantity = 0,
            ShipDate = "2024-11-06T13:14:15.369Z",
            Status = "placed",
            Complete = true
        };

        var createResponse = await _client.CreateOrderAsync(order);
        Assert.AreEqual(200, (int)createResponse.StatusCode, "Order creation failed");

        Thread.Sleep(1000); 

        var getResponse = await _client.GetOrderAsync(order.Id);
        Assert.AreEqual(200, (int)getResponse.StatusCode, "Order not found");

        var getOrder = JsonConvert.DeserializeObject<OrderModel>(await getResponse.Content.ReadAsStringAsync());
        Assert.AreEqual(order.Status, getOrder.Status, "Order status mismatch");
    }

    [TestMethod]
    [TestCategory("Order Management")]
    [Description("Delete an order")]
    public async Task TestDeleteOrder()
    {
        var order = new OrderModel
        {
            Id = 12,
            PetId = 0,
            Quantity = 0,
            ShipDate = "2024-11-06T13:14:15.369Z",
            Status = "placed",
            Complete = true
        };

        var createResponse = await _client.CreateOrderAsync(order);
        Assert.AreEqual(200, (int)createResponse.StatusCode, "Order creation failed");

        var deleteResponse = await _client.DeleteOrderAsync(order.Id);
        Assert.AreEqual(200, (int)deleteResponse.StatusCode, "Failed to delete order");

        var getResponse = await _client.GetOrderAsync(order.Id);
        Assert.AreEqual(404, (int)getResponse.StatusCode, "Order still exists after deletion");
    }
}